### PR TITLE
Tx pool start

### LINKED ISSUE #831: TxPool does not recovery correctly after node restart
There is code in txpool that saves txpool's state to DB and then recovers the txpool's from DB upon restart. The function that must restore txpool's state after restart does not work correctly, because all DB read methods that are used in it return nothing. They behave like nothing have been written.

One can use https://github.com/0xPolygonHermez/cdk-erigon/tree/tx-pool-start branch to simulate the issue.

There is a file zk/txpool/pool_zk_limbo_test.go
There are 2 methods Test_Persistency_Working and Test_Persistency_NotWorking. The only difference between these two methods is how they open the DB. The Test_Persistency_Working opens the DB in RW mode and everything works. The Test_Persistency_NotWorking opens the DB in RO mode and it DB does not return data at all. The second method simulates the problem that tx-pool has. So we must understand why it does not work and to fix it.

### PR DESCRIPTION
**Pull request by V-Staykov**

### CODE REVIEW

File: zk/txpool/pool_zk_limbo_test.go [hunk 1]
```diff
@@ -146,12 +200,12 @@ func initDb(t *testing.T, dbPath string) (kv.RwDB, kv.RwTx, kv.RwDB) {
...
 	txRw, err := database.BeginRw(ctx)
 	if err != nil {
-		t.Fatalf("Cannot craete db transaction")
+		t.Fatalf("Cannot craete db transaction %e", err)
 	}
 
 	aclDB, err := OpenACLDB(ctx, dbPath)
 	if err != nil {
-		t.Fatalf("Cannot craete acl db")
+		t.Fatalf("Cannot craete acl db %e", err)
```
Comment (revitteth) [inline zk/txpool/pool_zk_limbo_test.go]: Typo 'create'

File: zk/txpool/pool_zk_limbo_test.go [hunk 2]
```diff
@@ -146,12 +200,12 @@ func initDb(t *testing.T, dbPath string) (kv.RwDB, kv.RwTx, kv.RwDB) {
 
 	txRw, err := database.BeginRw(ctx)
 	if err != nil {
-		t.Fatalf("Cannot craete db transaction")
+		t.Fatalf("Cannot craete db transaction %e", err)
```
Comment (revitteth) [inline zk/txpool/pool_zk_limbo_test.go]: Typo 'create'

**Review by revitteth — DISMISSED** [review]

File: zk/txpool/pool_zk_limbo.go [hunk 3]
```diff
@@ -532,9 +532,9 @@ func (p *TxPool) fromDBLimbo(ctx context.Context, tx kv.Tx, cacheView kvcache.Ca
 			txn.SenderID, txn.Traced = p.senders.getOrCreateID(addr)
 			binary.BigEndian.Uint64(v)
 
-			if reason := p.validateTx(txn, true, cacheView, addr); reason != NotSet && reason != Success {
-				return nil
-			}
+			// if reason := p.validateTx(txn, true, cacheView, addr); reason != NotSet && reason != Success {
```
Comment (mandrigin) [inline zk/txpool/pool_zk_limbo.go:540]: why is this commented out?

**Review by mandrigin — CHANGES_REQUESTED** [review]
1 question

File: zk/txpool/pool_zk_limbo.go (hunk 3 above)
Comment (revitteth) [reply zk/txpool/pool_zk_limbo.go:540]: I think @kstoykov knows why this is commented

Comment (V-Staykov) [reply zk/txpool/pool_zk_limbo.go:540]: @kstoykov can comment on this

Comment (kstoykov) [reply zk/txpool/pool_zk_limbo.go:540]: ValidateTx function validates a tx against current network state. Limbo transactions are expected to be invalid according to current network state. That's why there is no point to check it while recovering the pool from a database. These transactions may become valid after some of the current tx in the pool are executed so leave the decision whether a limbo transaction (or any other transaction that has been unwound) to the execution stage.

Comment (mandrigin) [reply zk/txpool/pool_zk_limbo.go:540]: fair enough, but better to add a comment there and remoce the code completely, then it will be okay.

**Review by mandrigin — APPROVED** [review]

**Pull request merged**